### PR TITLE
Add `ignoreFunctions` option to `function-no-unknown`

### LINKED ISSUE #5900: Add `ignoreFunctions: []` to function-no-unknown
### What is the problem you're trying to solve?

[function-no-unknown](https://stylelint.io/user-guide/rules/list/function-no-unknown) throws an error when using non-standard funcitons such as tailwindcss [theme](https://tailwindcss.com/docs/functions-and-directives#theme) function.

### What solution would you like to see?

It would be nice to have a feature to ignore certain functions just like `ignoreAtRules` of [at-rule-no-unknown](https://stylelint.io/user-guide/rules/list/at-rule-no-unknown).

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Closes #5900

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: lib/rules/function-no-unknown/README.md [hunk 1]
```diff
@@ -35,3 +35,46 @@ a { transform: scale(1); }
...
+<!-- prettier-ignore -->
+```css
+a { transform: THEME(1); }
+```
+
+<!-- prettier-ignore -->
+```css
+a { transform: foo(1); }
+```
+
```
Comment (jeddy3) [inline lib/rules/function-no-unknown/README.md:69]: Let's remove this section.

For `ignore` options we usually just show the rule becoming more permissive in the docs.

File: lib/rules/function-no-unknown/index.js [hunk 2]
```diff
@@ -23,14 +25,34 @@ const meta = {
...
 
+		/** @type {Array<string | RegExp>} */
+		const ignoreFunctions = (secondaryOptions && secondaryOptions.ignoreFunctions) || [];
+
+		/** @type {(funcName: string) => boolean} */
+		const isFunctionIgnored =
+			ignoreFunctions.length === 0
+				? () => false
+				: (funcName) => Boolean(matchesStringOrRegExp(funcName, ignoreFunctions));
+
```
Comment (jeddy3) [inline lib/rules/function-no-unknown/index.js:55]: If possible, let's use the [`optionsMatches`](https://github.com/stylelint/stylelint/blob/main/lib/utils/optionsMatches.js) util here.

**Review by jeddy3 — CHANGES_REQUESTED** [review]
Thanks for getting started on this so quickly!

I've made two suggestions.

File: lib/rules/function-no-unknown/index.js (hunk 2 above)
Comment (ybiquitous) [reply lib/rules/function-no-unknown/index.js]: Thanks! I forgot the util. 😅

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you!

**Comment by ybiquitous** [comment]
@jeddy3 I have no left for this pull request. Can we merge this?

**Comment by jeddy3** [comment]
@ybiquitous Yes, go ahead.

I've changed the branch protection rules to require one approval.  It's a more sensible default. We can reserve two reviews for only the largest pull requests.

**Comment by ybiquitous** [comment]
Thank you!

**Pull request merged**